e2e_tests/tests/fixtures/custom_searcher/model_noop.py

```diff
@@ -0,0 +1,259 @@
+import collections
+import json
+import logging
+import os
+import pathlib
+import pickle
+import random
+import sys
+import time
+from typing import Any, Dict, Optional, Type
+
+import numpy as np
+
+import determined as det
+from determined import layers, tensorboard, util, workload
+from determined.common import check
+
+
+class NoOpTrialContext(det.TrialContext):
+    """
+    NoOpTrial needs batch sizes.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._per_slot_batch_size, self._global_batch_size = util.calculate_batch_sizes(
+            self.get_hparams(),
+            self.env.experiment_config.slots_per_trial(),
+            "NoOpTrial",
+        )
+
+    def get_per_slot_batch_size(self) -> int:
+        return self._per_slot_batch_size
+
+    def get_global_batch_size(self) -> int:
+        return self._global_batch_size
+
+
+class NoOpTrialController(det.TrialController):
```

Comment:
This model_noop.py has way too much cruft to just copy/paste it.

The real NoOpTrialController is used in like 30 different tests to obtain like 30 different behaviors.  It's also one of two existing non-`determined` TrialController clases, which are overall going away.  All changes made to the to the TrialController classes have to get copied into the existing NoOpTrialController and the MetricMaker TrialController.  This is a pain in the butt, and copy/pasting NoOpTrialController compounds this pain.

Instead, can you write a core api training loop?  I think it will amount to far less code and it shouldn't increase our developer burden any.